lib/zold/wallet.rb

```diff
@@ -121,6 +121,7 @@ def sub(amount, invoice, pvt, details = '-', time: Time.now)
     end
 
     def add(txn)
+      raise "Wallet amount will exceed MAX if applied #{txn}" if balance.to_i + txn.amount.to_i > Amount::MAX
```

Comment:
@liligga I’d suggest extracting it out. We shouldn’t need to update the same logic in 2 places.

Reply:
@botanicus do you suggest some kind of error subclass class?

Reply:
@liligga I think that we should use `balance + txn.amount > Amount::MAX`. There is no point in converting them to Integer.

Reply:
@liligga private method would be enough. The condition logic is the same, extracting a subclass wouldn't address it (on its own).

Reply:
@botanicus did you mean protected method to put in `module Zold`?

Reply:
@yegor256 I think there is a point in converting. They are `Amount` but `Amount::MAX` is an integer so it has to be either `Amount.new(Amount::MAX)` or these two with `.to_i`.

Reply:
@liligga something along the lines could do (in my view, @yegor256's opinion might differ on this).